### PR TITLE
Added new assets endpoints

### PR DESCRIPTION
**Pull request by dsfish**
This PR adds support for the following new endpoints:
- /asset_report/audit_copy/get
- /asset_report/filter
- /asset_report/refresh

It also adds test coverage for each.

### CODE REVIEW

File: lib/plaid/products/asset_report.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ class AssetReport < BaseProduct
     #                  the Asset Report. Plaid will return as much data as
     #                  possible within the implied date range, up to a maximum
     #                  of 730 days (2 years) of history.
-    # options -        An optional object containing client-provided user
+    # options        - An optional object containing client-provided user
```
Comment (be9) [inline lib/plaid/products/asset_report.rb]: Let's make `options` really optional by specifying `, options={}` in the method definition.

File: lib/plaid/products/asset_report.rb [hunk 2]
```diff
@@ -23,6 +23,43 @@ def create(access_tokens, days_requested, options)
...
+    #                      in the Asset Report. Plaid will return as much data
+    #                      as possible within the implied date range, up to a
+    #                      maximum of 730 days (2 years) of history.
+    # options            - An optional object containing client-provided user
+    #                      information, a client-provided report ID, and a
+    #                      webhook. For more information, see the website
+    #                      listed above.
+    #
+    # Returns a AssetReportRefreshResponse object.
+    def refresh(asset_report_token, days_requested, options)
```
Comment (be9) [inline lib/plaid/products/asset_report.rb]: Same idea with `options={}`.

File: lib/plaid/products/asset_report.rb [hunk 3]
```diff
@@ -94,6 +143,14 @@ class AssetReportCreateResponse < Models::BaseResponse
     property :asset_report_id
   end
 
+  # Public: The response wrapper for /asset_report/filter.
+  class AssetReportFilterResponse < AssetReportCreateResponse
```
Comment (be9) [inline lib/plaid/products/asset_report.rb]: Do we have any known fields here?

File: lib/plaid/products/asset_report.rb [hunk 4]
```diff
@@ -94,6 +143,14 @@ class AssetReportCreateResponse < Models::BaseResponse
...
+  end
+
+  # Public: The response wrapper for /asset_report/refresh.
+  class AssetReportRefreshResponse < AssetReportCreateResponse
+  end
```
Comment (be9) [inline lib/plaid/products/asset_report.rb]: Do we have any known fields here?

File: lib/plaid/products/asset_report.rb [hunk 5]
```diff
@@ -115,15 +172,19 @@ class AssetReportRemoveResponse < Models::BaseResponse
...
     ##
     # :attr_reader:
     # Public: The audit copy token.
     property :audit_copy_token
   end
 
-  # Public: The response wrapper for /audit_copy/remove.
+  # Public: The response wrapper for /asset_report/audit_copy/get.
+  class AuditCopyGetResponse < AssetReportGetResponse
+  end
```
Comment (be9) [inline lib/plaid/products/asset_report.rb]: Do we have any known fields here?

File: lib/plaid/products/asset_report.rb (hunk 3 above)
Comment (dsfish) [reply lib/plaid/products/asset_report.rb]: I'm not sure what you mean. `AssetReportFilterResponse` is the same as `AssetReportCreateResponse`. Is there a better way to represent this?

File: lib/plaid/products/asset_report.rb (hunk 4 above)
Comment (dsfish) [reply lib/plaid/products/asset_report.rb]: See above

File: lib/plaid/products/asset_report.rb (hunk 5 above)
Comment (dsfish) [reply lib/plaid/products/asset_report.rb]: See above

File: lib/plaid/products/asset_report.rb (hunk 1 above)
Comment (dsfish) [reply lib/plaid/products/asset_report.rb]: ⚡️

File: lib/plaid/client.rb [hunk 6]
```diff
@@ -8,6 +8,9 @@ class Client
     # Public: The current environment in use (one of ENVIRONMENTS).
     attr_reader :env
 
+    # Public: The client ID.
+    attr_accessor :client_id
```
Comment (be9) [inline lib/plaid/client.rb]: BTW if you don't want `client_id` to be changed, let's use `attr_reader`.

File: lib/plaid/products/asset_report.rb (hunk 3 above)
Comment (be9) [reply lib/plaid/products/asset_report.rb]: I see. Just missed the fact that `AssetReportCreateResponse` already has fields defined.

**Pull request merged**